src/connection.rs

```diff
@@ -37,14 +37,79 @@ pub enum Error {
 
 pub type Result<T, E = Error> = std::result::Result<T, E>;
 
+/// How to connect to a `Transport`
```

Comment:
If you connect to a "transport", then why is `async fn connection` returning `R: Request`? Is this a bit of a naming clash because `Transport` was already taken by `crate::connection::transport::Transport`? Maybe we can call `Request` a `RequestHandler`. I think nouns are better for traits and structs and verbs rather for functions. Same goes for `Connect::connection` which I would rather call `ConnectionHandler::connect`, esp. since the method actually establishes a new connection rather than using a cached one (we currently don't cache connections at this specific layer and I think we don't want to because Kafka doesn't really multiplex TCP connection requests but only allows in-order pipelining).

Reply:
Yup, these names were not great. I've taken your suggestions here, how does it look now?